packages/javascript/bh-shared-ui/src/components/ExploreTable/ExploreTable.tsx

```diff
@@ -14,109 +14,201 @@
-import { makeFormattedObjectInfoFieldsMap } from '../../utils';
+import { GraphNode } from 'js-client-library';
+import { ChangeEvent, memo, useCallback, useMemo, useState } from 'react';
+import { useToggle } from '../../hooks';
+import { WrappedExploreTableItem } from '../../types';
+import { EntityField, format, formatPotentiallyUnknownLabel } from '../../utils';
 import NodeIcon from '../NodeIcon';
-import { TableControls } from './TableControls';
+import { ManageColumnsComboBoxOption } from './ManageColumnsComboBox/ManageColumnsComboBox';
+import TableControls from './TableControls';
 
-type HasData = { data?: object };
+const REQUIRED_EXPLORE_TABLE_COLUMN_KEYS = ['nodetype', 'objectid', 'displayname'];
 
-interface ExploreTableProps<TData extends HasData> {
+const requiredColumns = REQUIRED_EXPLORE_TABLE_COLUMN_KEYS.reduce(
+    (acc, curr) => ({ ...acc, [curr]: true }),
+    {}
+) as Record<string, boolean>;
+
+type MungedTableRowWithId = WrappedExploreTableItem['data'] & { id: string };
+
+const columnhelper = createColumnHelper();
+
+interface ExploreTableProps {
     open?: boolean;
     onClose?: () => void;
-    data?: Record<string, TData>;
+    data?: Record<string, WrappedExploreTableItem>;
+    selectedColumns?: Record<string, boolean>;
+    allColumnKeys?: string[];
+    onManageColumnsChange?: (columns: ManageColumnsComboBoxOption[]) => void;
 }
 
-const ExploreTable = <TData extends HasData>({ data, open, onClose }: ExploreTableProps<TData>) => {
+const MemoDataTable = memo(DataTable);
+
+const makeColumnDef = (key: string) =>
+    columnhelper.accessor(key, {
+        header: formatPotentiallyUnknownLabel(key),
+        cell: (info) => {
+            const value = info.getValue() as EntityField['value'];
+
+            if (typeof value === 'boolean') {
+                return value ? (
+                    <div className='h-full w-full flex justify-center items-center text-center'>
+                        <FontAwesomeIcon icon={faCheck} color='green' className='scale-125' />{' '}
+                    </div>
+                ) : (
+                    <div className='h-full w-full flex justify-center items-center text-center'>
+                        <FontAwesomeIcon icon={faCancel} color='lightgray' className='scale-125' />{' '}
+                    </div>
+                );
+            }
+
+            return format({ keyprop: key, value, label: key }) || '--';
+        },
+        id: key,
+    });
+
+const ExploreTable = ({
+    data,
+    open,
+    onClose,
+    onManageColumnsChange,
+    allColumnKeys,
+    selectedColumns,
+}: ExploreTableProps) => {
     const [searchInput, setSearchInput] = useState('');
+    const [isExpanded, toggleIsExpanded] = useToggle(false);
+
     const mungedData = useMemo(
-        () => (data && Object.keys(data).map((id) => ({ ...data?.[id]?.data, id }))) || [],
+        () =>
+            // TODO: remove id and just use objectid for onRowClick/getRowId?
```

Comment:
this should work because the `selectedItem` param works with a db id or objectid!